sqlmesh/core/snapshot/evaluator.py

```diff
@@ -449,13 +449,14 @@ def _demote_snapshot(
     def _cleanup_snapshot(self, snapshot: SnapshotInfoLike) -> None:
         snapshot = snapshot.table_info
         table_names = [snapshot.table_name()]
-        if snapshot.version != snapshot.fingerprint:
+        if snapshot.version != snapshot.fingerprint.to_version():
```

Comment:
@izeigerman I think this has been bugged for a while. The bug would result in this always being true and therefore always adding the table name. Wondering if I am missing something and if we really expect it to behave the way I just fixed it to behave.

Reply:
it's because fingerprint didn't used to be an object probably

Reply:
Yeah this code does look old so that makes sense